build/source/engine/paramCheck.f90

```diff
@@ -109,27 +110,27 @@ subroutine paramCheck(err,message)
  ! ************************************
 
  ! check that the maximum transpiration limit is within bounds
- if(mpar_data%var(iLookPARAM%critSoilTranspire)>mpar_data%var(iLookPARAM%theta_sat) .or. &
-    mpar_data%var(iLookPARAM%critSoilTranspire)<mpar_data%var(iLookPARAM%theta_res))then
+ if(mpar_data(iLookPARAM%critSoilTranspire)>mpar_data(iLookPARAM%theta_sat) .or. &
+    mpar_data(iLookPARAM%critSoilTranspire)<mpar_data(iLookPARAM%theta_res))then
   message=trim(message)//'critSoilTranspire parameter is out of range '// &
                          '[NOTE: if overwriting Noah-MP soil table values in paramTrial, must overwrite all soil parameters]'
   err=20; return
  endif
 
  ! check that the soil wilting point is within bounds
- if(mpar_data%var(iLookPARAM%critSoilWilting)>mpar_data%var(iLookPARAM%theta_sat) .or. &
-    mpar_data%var(iLookPARAM%critSoilWilting)<mpar_data%var(iLookPARAM%theta_res))then
-  print*, 'mpar_data%var(iLookPARAM%theta_res) = ', mpar_data%var(iLookPARAM%theta_res)
-  print*, 'mpar_data%var(iLookPARAM%theta_sat) = ', mpar_data%var(iLookPARAM%theta_sat)
+ if(mpar_data(iLookPARAM%critSoilWilting)>mpar_data(iLookPARAM%theta_sat) .or. &
+    mpar_data(iLookPARAM%critSoilWilting)<mpar_data(iLookPARAM%theta_res))then
+  print*, 'mpar_data(iLookPARAM%theta_res) = ', mpar_data(iLookPARAM%theta_res)
```

Comment:
We should consider adding logging functions as in VIC. That will allow us to control these messages better (and send them to different places).